core/include/traccc/clusterization/detail/sparse_ccl.hpp

```diff
@@ -24,12 +27,18 @@ namespace detail {
 /// @param L an equivalance table
 ///
 /// @return the root of @param e
-template <template <typename> class vector_t>
-TRACCC_HOST_DEVICE inline unsigned int find_root(
-    const vector_t<unsigned int>& L, const unsigned int& e) {
+TRACCC_HOST_DEVICE inline unsigned int find_root(const unsigned int* L,
+                                                 unsigned int e,
+                                                 unsigned int size) {
```

Comment:
This I don't like - we should not be introducing additional parameters simply for debugging assertions. Instead, we should turn `L` back into a vector reference, which will also give us the size without passing it separately.

Reply:
Moved the code to using `vecmem::device_vector<unsigned int>`...